diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/BooleanExpressionsRule.kt

```diff
@@ -0,0 +1,165 @@
+package org.cqfn.diktat.ruleset.rules.chapter3
+
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings.COMPLEX_BOOLEAN_EXPRESSION
+import org.cqfn.diktat.ruleset.rules.DiktatRule
+import org.cqfn.diktat.ruleset.utils.KotlinParser
+import org.cqfn.diktat.ruleset.utils.findAllNodesWithCondition
+import org.cqfn.diktat.ruleset.utils.findLeafWithSpecificType
+
+import com.bpodgursky.jbool_expressions.Expression
+import com.bpodgursky.jbool_expressions.parsers.ExprParser
+import com.bpodgursky.jbool_expressions.rules.RuleSet
+import com.pinterest.ktlint.core.ast.ElementType
+import com.pinterest.ktlint.core.ast.ElementType.BINARY_EXPRESSION
+import com.pinterest.ktlint.core.ast.ElementType.OPERATION_REFERENCE
+import com.pinterest.ktlint.core.ast.ElementType.PARENTHESIZED
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+
+import java.lang.RuntimeException
+
+/**
+ * Rule that warns if the boolean expression can be simplified.
+ */
+class BooleanExpressionsRule(configRules: List<RulesConfig>) : DiktatRule(
+    "boolean-expressions-rule",
+    configRules,
+    listOf(COMPLEX_BOOLEAN_EXPRESSION)) {
+    override fun logic(node: ASTNode) {
+        if (node.elementType == ElementType.CONDITION) {
+            checkBooleanExpression(node)
+        }
+    }
+
+    @Suppress("TooGenericExceptionCaught")
+    private fun checkBooleanExpression(node: ASTNode) {
+        // This map is used to represent expressions as single chars, which are used by the lib.
```

Comment:
I'd say 'Assign a variable name for every elementary boolean expression', from your comment I have originally thought that library requires some weird encoding of boolean expression, but actually it's quite natural

Reply:
fixed